Add random ray support (#1305)
neams-th-coe/cardinal: doc/content/source/problems/OpenMCCellAverageProblem.md

with `ifp_generations` (an extra double and an extra integrer must be tracked per particle
for each generation). It is recommend that the number of IFP generations be minimized when possible.

## Random Ray Neutronics
  id=random_ray

Cardinal supports solving both fixed source and k-eigenvalue neutron transport problems with
[!ac](TRRM) in OpenMC. First-class support is provided for temperature feedback, density feedback,
and deforming geometry when using [!ac](TRRM). When using density feedback with either [!ac](TRRM)
or multi-group Monte Carlo, `mgxs_reference_densities` must be provided. This is required to

> convert the field variables provided (which have units of kg m^-3^) to the unitless density

Reviewer: ```suggestion
convert the field variables provided (which have units of kg m$^{-3}$) to the unitless density
```

> multiplier expected by OpenMC. This should be the same density that was used to generate

Reviewer 2: ```suggestion
multiplier expected by OpenMC. The values used in `mgxs_reference_densities` should be the same density values that were used to generate
```

the [!ac](MGXS) for the materials filling the cells where density feedback is applied.

> `OpenMCCellAverageProblem` allows for the control of the inactive and active ray length

Reviewer 2: Instead of having this sentence here about some XML parameters that can be overridden via Cardinal, let's move it to the `Controlling the OpenMC Settings` section later in this file since it's IMO more aligned with that part.

from the Cardinal input file by setting `inactive_distance` and `active_distance` (overriding the
values provided in the `settings.xml` file). An `OpenMCCellAverageProblem` block using [!ac](TRRM)
will look quite similar to the equivalent continuous energy simulation,

!listing test/tests/neutronics/mg/rr_doppler_slab_root/openmc.i
  block=Problem

Several features supported in continuous energy Monte Carlo neutron transport are not supported when
using [!ac](TRRM). These include the calculation of kinetics parameters, the use of tally/filter/material

> editors, and the [BoratedWater](BoratedWater.md) criticality search. Finally, tally/filter/score support is limited to

Reviewer 2: ```suggestion
editors, and some criticality searches. Finally, tally/filter/score support is limited to
```

Reviewer 2: less prone to needing an update later on if there's other criticality search objects which are not applicable with random ray
